electronicparsers/openmx/parser.py

```
@@ -767,6 +809,35 @@ def parse(self, mainfile: str, archive: EntryArchive, logger):
             ignore_md_file = True
             self.logger.warning('.md file does not contain enough MD steps')
 
+        # Stress tensor values are not written to outfile but only to stdout.
+        # That means we have to check if the stdout is actually present and
+        # look for the stress values there.
+        stress_tensor_enabled = mainfile_parser.get('scf.stress.tensor')
+        stdout_uele_list = []
+        stress_tensors = None
+        if stress_tensor_enabled == 'on':
+            # find_stdout returns path to correct stdout if it exists
+            # or None otherwise
+            stdout_path = find_stdout(mainfile)
+
+            if mainfile_md_steps is not None and stdout_path is not None:
```

> if mainfile_md_steps is not None

This logic kinda comes out of the blue... Could you add context as a comment?

---

Yeah, this was halfway baked check to not except later when we index the `mainfile_md_steps`, but we were still missing checks that we actually have scf steps, and more... 
In the end I just wrapped all of this instead in `try: .... except: pass` block instead of the `if mainfile_md_steps is not None and stdout_path is not None:` check.